File: java/weak_cryptography/Encryption.java
import javax.crypto.Cipher;
import javax.crypto.SecretKey;
import javax.crypto.spec.SecretKeySpec;
import java.util.Base64;

public class Encryption {

    // VULN 1: DES cipher used to encrypt stored payment card data
    public String encryptCardData(String cardNumber, byte[] desKey) throws Exception {
        SecretKey key = new SecretKeySpec(desKey, "DES");
        Cipher cipher = Cipher.getInstance("DES");
        cipher.init(Cipher.ENCRYPT_MODE, key);
        byte[] encrypted = cipher.doFinal(cardNumber.getBytes());
        return Base64.getEncoder().encodeToString(encrypted);
    }

    // VULN 2: DES in ECB mode used to encrypt session tokens
    public byte[] encryptSessionData(byte[] sessionPayload, byte[] desKey) throws Exception {
        SecretKey key = new SecretKeySpec(desKey, "DES");
        Cipher cipher = Cipher.getInstance("DES/ECB/PKCS5Padding");
        cipher.init(Cipher.ENCRYPT_MODE, key);
        return cipher.doFinal(sessionPayload);
    }

    // VULN 3: AES in ECB mode (no IV) used to encrypt order details at rest
    public String encryptOrderData(String orderJson, byte[] aesKey) throws Exception {
        SecretKey key = new SecretKeySpec(aesKey, "AES");
        Cipher cipher = Cipher.getInstance("AES/ECB/PKCS5Padding");
        cipher.init(Cipher.ENCRYPT_MODE, key);
        byte[] encrypted = cipher.doFinal(orderJson.getBytes());
        return Base64.getEncoder().encodeToString(encrypted);
    }
}

File: kotlin/weak_cryptography/Encryption.kt
import javax.crypto.Cipher
import javax.crypto.spec.SecretKeySpec
import java.util.Base64

class Encryption {
    private val desKey = "A1B2C3D4".toByteArray(Charsets.UTF_8)
    private val rc2Key = "MyRC2Key12345678".toByteArray(Charsets.UTF_8)
    private val aesKey = "AES128BitKey1234".toByteArray(Charsets.UTF_8)

    // VULN 1: DES/ECB encryption for payment card data
    fun encryptCardData(cardNumber: String): String {
        val cipher = Cipher.getInstance("DES/ECB/PKCS5Padding")
        val keySpec = SecretKeySpec(desKey, "DES")
        cipher.init(Cipher.ENCRYPT_MODE, keySpec)
        val encrypted = cipher.doFinal(cardNumber.toByteArray(Charsets.UTF_8))
        return Base64.getEncoder().encodeToString(encrypted)
    }

    fun decryptCardData(encryptedCard: String): String {
        val cipher = Cipher.getInstance("DES/ECB/PKCS5Padding")
        val keySpec = SecretKeySpec(desKey, "DES")
        cipher.init(Cipher.DECRYPT_MODE, keySpec)
        val decrypted = cipher.doFinal(Base64.getDecoder().decode(encryptedCard))
        return String(decrypted, Charsets.UTF_8)
    }

    // VULN 2: RC2/ECB encryption for session data
    fun encryptSessionData(sessionPayload: String): String {
        val cipher = Cipher.getInstance("RC2/ECB/PKCS5Padding")
        val keySpec = SecretKeySpec(rc2Key, "RC2")
        cipher.init(Cipher.ENCRYPT_MODE, keySpec)
        val encrypted = cipher.doFinal(sessionPayload.toByteArray(Charsets.UTF_8))
        return Base64.getEncoder().encodeToString(encrypted)
    }

    // VULN 3: AES/ECB (no IV) for PII data - customer personal info
    fun encryptPii(piiData: String): String {
        val cipher = Cipher.getInstance("AES/ECB/PKCS5Padding")
        val keySpec = SecretKeySpec(aesKey, "AES")
        cipher.init(Cipher.ENCRYPT_MODE, keySpec)
        val encrypted = cipher.doFinal(piiData.toByteArray(Charsets.UTF_8))
        return Base64.getEncoder().encodeToString(encrypted)
    }

    fun decryptPii(encryptedPii: String): String {
        val cipher = Cipher.getInstance("AES/ECB/PKCS5Padding")
        val keySpec = SecretKeySpec(aesKey, "AES")
        cipher.init(Cipher.DECRYPT_MODE, keySpec)
        val decrypted = cipher.doFinal(Base64.getDecoder().decode(encryptedPii))
        return String(decrypted, Charsets.UTF_8)
    }
}
